lib/taurus/tauruscustomsettings.py

```diff
@@ -68,6 +68,14 @@
 # providing support to new schemes
 # EXTRA_SCHEME_MODULES = ['myownschememodule']
 
+# Set a custom formatter. Taurus widgets use a default formatter based on the
+# attribute type, but sometimes a custom formatter is needed.
+# The formatter can be a python format string or formatter callable
+# (in string version)
+#e.g.
+# DEFAULT_FORMATTER = '{0}'
+# DEFAULT_FORMATTER = 'taurus.core.tango.util.formatter.tangoFormatter'
+
```

Comment:
Consider adding a `--default-formatter` option in the argument parser of TaurusApplication which would override `tauruscustomsettings.DEFAULT_FORMATTER`

Reply:
Agree with the `--default-formatter` option for individual widgets (form, plots, ...) 
but, what does it mean override for you? More precedence? or replace it for all the widgets? 

I think it is difficult to  manage in complex GUI e.g.  taurusgui where it should affect to all widgets

But IMHO it is a extra since it can be done by a menu action.

Reply:
Implementationwise I was thinking that --default-formatter would simply overwrite at runtime the`tauruscustomsettings.DEFAULT_FORMATTER` value

Then any widget in the app which does not explicitly set a formatter, would use this one. If an app needs finer control, they should neither be using this parameter nor the setting.